rust/fory-core/src/fory.rs

```diff
@@ -555,6 +554,7 @@ impl Fory {
         record: &T,
         context: &mut WriteContext,
     ) -> Result<Vec<u8>, Error> {
+        context.writer.attach_buffer(vec![]);
```

Comment:
This will allocate a new buffer everytime, which cause extra copy and allocation when serializing